first-commits: initialize ami with base Windows Server 2022 (#10)
cisagov/windows-server-packer: README.md

> # windows-server-packer 💀📦 #

Reviewer: ```suggestion
# windows-server-packer 🪟📦 #
```

Author: @mcdonnnj resolved with [547d1e7](https://github.com/cisagov/windows-server-packer/pull/10/commits/547d1e7a954fcafb2312866edcefcc4a816514fe)

[![GitHub Build Status](https://github.com/cisagov/windows-server-packer/workflows/build/badge.svg)](https://github.com/cisagov/windows-server-packer/actions)

> This project can be used to create a custom Amazon Machine Images based on

Reviewer: ```suggestion
This project can be used to create a custom Amazon Machine Image based on
```

Author: resolved with [313fb34](https://github.com/cisagov/windows-server-packer/pull/10/commits/313fb34d0568f9f50a9fc78cf03b29baa2dbdc26)

> Windows Server 2022

Reviewer: ```suggestion
Windows Server 2022.
```

Author: @mcdonnnj resolved with [ed022ff](https://github.com/cisagov/windows-server-packer/pull/10/commits/ed022ffe3bd5914a5ec4f1842aff56fceb499a6f)

[![GitHub Build Status](https://github.com/cisagov/windows-commando-packer/workflows/build/badge.svg)](https://github.com/cisagov/windows-commando-packer/actions)

> This project can be used to create machine images based on

Reviewer: ```suggestion
This project can be used to create Amazon Machine Images based on
```

Author: resolved with this commit: [78420ea](https://github.com/cisagov/windows-commando-packer/pull/10/commits/78420ea80f0d049af5aa0e10dd34b6abde403aa6)

| release\_tag | The GitHub release tag to use for the tags applied to the created AMI. | `string` | `""` | no |
| release\_url | The GitHub release URL to use for the tags applied to the created AMI. | `string` | `""` | no |
| skip\_create\_ami | Indicate if Packer should not create the AMI. | `bool` | `false` | no |

> | winrm\_password | The password used to connect to the instance via WinRM. | `string` | `false` | no |

Reviewer: Just for documentation we'll present this as a required variable.

```suggestion
| winrm\_password | The password used to connect to the instance via WinRM. | `string` | n/a | yes |
```

Author: @mcdonnnj resolved with [2973420](https://github.com/cisagov/windows-server-packer/pull/10/commits/29734201bc39508d11674b968a889785ba6b5cb6)

```console
pip install --requirement requirements-dev.txt

> AWS_PROFILE=cool-images-ec2amicreate-windows-server-packer packer build --timestamp-ui -var release_tag=$(./bump_version.sh show) -var is_prerelease=true src/packer.pkr.hcl

Reviewer: The example build command should be updated to reflect that you _must_ pass a value for `winrm_password`.

Author: @mcdonnnj resolved with [a7dae0d](https://github.com/cisagov/windows-server-packer/pull/10/commits/a7dae0df1dc23fa92edf9e0000d303e6582d7a8b)

Reviewer: Please update this to pass the argument on the command-line instead of as an environment variable.

Author: @mcdonnnj resolved with [46c6ab2](https://github.com/cisagov/windows-server-packer/pull/10/commits/46c6ab255bd389117c5663ab2b288b092cf80c4f)

```console
pip install --requirement requirements-dev.txt

> ansible-galaxy install --force --force-with-deps --role-file src/requirements.yml

Reviewer: You've removed Ansible logic from this project.

```suggestion
```

Author: resolved with [313fb34](https://github.com/cisagov/windows-server-packer/pull/10/commits/313fb34d0568f9f50a9fc78cf03b29baa2dbdc26)

```console
pip install --requirement requirements-dev.txt
AWS_PROFILE=cool-images-ec2amicreate-windows-server-packer \

> packer build --timestamp-ui -var release_tag=$(./bump_version.sh show) -var is_prerelease=true -var winrm_password={password} src/packer.pkr.hcl

Reviewer 2: Since this is intended to document an example, can you please change this slightly:
```suggestion
packer build --timestamp-ui -var release_tag=$(./bump_version.sh show) \
-var is_prerelease=true -var winrm_password="your-winrm-password" src/packer.pkr.hcl
```

Note that I also added a line-continuation character to break up that long line.

Author: @dav3r resolved with [09eb2e6](https://github.com/cisagov/windows-server-packer/pull/10/commits/09eb2e6852084dffa68fddd605f151664e95cc86)

Reviewer 2: @itsmostafa Please check again; https://github.com/cisagov/windows-server-packer/pull/10/commits/09eb2e6852084dffa68fddd605f151664e95cc86 does not change the example password from `{password}` to `"your-winrm-password"`.

Author: @dav3r i missed this instance. resolved with [765a968](https://github.com/cisagov/windows-server-packer/pull/10/commits/765a968430a18b95b4fbd04acc92742dd01778b4)

Reviewer 2: All good now, thanks!

```console
pip install --requirement requirements-dev.txt
AWS_PROFILE=cool-images-ec2amicreate-windows-server-packer \
packer build --timestamp-ui -var release_tag=$(./bump_version.sh show) \

> -var is_prerelease=true -var winrm_password="your-winrm-password src/packer.pkr.hcl

Reviewer 2: You missed the closing quotes here:
```suggestion
-var is_prerelease=true -var winrm_password="your-winrm-password" src/packer.pkr.hcl
```

Author: @dav3r oops.. resolved with [04bb9a9](https://github.com/cisagov/windows-server-packer/pull/10/commits/04bb9a965b7f111354b020e74618ca4b0c46209b)

```

```console

> AWS_PROFILE=cool-images-ec2amicreate-windows-server-packer packer build --timestamp-ui -var-file release.pkrvars.hcl src/packer.pkr.hcl

Reviewer: The example build command should be updated to reflect that you _must_ pass a value for `winrm_password`.

Author: @mcdonnnj resolved with [a7dae0d](https://github.com/cisagov/windows-server-packer/pull/10/commits/a7dae0df1dc23fa92edf9e0000d303e6582d7a8b)

Reviewer: Please update this to pass the argument on the command-line instead of as an environment variable.

Author: @mcdonnnj resolved with [46c6ab2](https://github.com/cisagov/windows-server-packer/pull/10/commits/46c6ab255bd389117c5663ab2b288b092cf80c4f)

```

```console

> PKR_VAR_winrm_password="your-winrm-password" \
> AWS_PROFILE=cool-images-ec2amicreate-windows-server-packer \
> packer build --timestamp-ui -var winrm_password={password} \ -var-file release.pkrvars.hcl src/packer.pkr.hcl

Reviewer 2: Similar change as above, except I removed the now-unnecessary `PKR_VAR_winrm_password` env variable, which I assume was an oversight in https://github.com/cisagov/windows-server-packer/pull/10/commits/46c6ab255bd389117c5663ab2b288b092cf80c4f:
```suggestion
AWS_PROFILE=cool-images-ec2amicreate-windows-server-packer \
packer build --timestamp-ui -var winrm_password="your-winrm-password" \
-var-file release.pkrvars.hcl src/packer.pkr.hcl
```

Author: @dav3r wasn't sure how I should represent an example password. I went with your suggestion. Resolved with [09eb2e6](https://github.com/cisagov/windows-server-packer/pull/10/commits/09eb2e6852084dffa68fddd605f151664e95cc86)

Reviewer 2: @itsmostafa Please check again; https://github.com/cisagov/windows-server-packer/pull/10/commits/09eb2e6852084dffa68fddd605f151664e95cc86 does not remove this line (which it should):
```
PKR_VAR_winrm_password="your-winrm-password" \
```

Author: @dav3r resolved with [765a968](https://github.com/cisagov/windows-server-packer/pull/10/commits/765a968430a18b95b4fbd04acc92742dd01778b4)

Reviewer 2: This looks good now, thanks!

```

```console
AWS_PROFILE=cool-images-ec2amicreate-windows-server-packer \
packer build --timestamp-ui -var winrm_password="your-winrm-password" \

> -var-file release.pkrvars.hcl src/packer.pkr.hcl

Reviewer: Please move the `winrm_password` definition into the example `release.pkrvars.hcl` file above.

```suggestion
AWS_PROFILE=cool-images-ec2amicreate-windows-server-packer packer build --timestamp-ui -var-file release.pkrvars.hcl src/packer.pkr.hcl
```

Author: @mcdonnnj resolved with [bd38a30](https://github.com/cisagov/windows-server-packer/pull/10/commits/bd38a30386e60e51b74bd3435da3359d2790e517)

Reviewer: @itsmostafa You removed the `-var` argument but you did not include a variable definition in the `release.pkrvars.hcl` example:
https://github.com/cisagov/windows-server-packer/blob/2facb314a95463b8478e01f6b5e40410842c104d/README.md?plain=1#L138-L145

Author: @mcdonnnj resolved with [93c9408](https://github.com/cisagov/windows-server-packer/pull/10/commits/93c9408a606bfff406ff49b76c0acd9e6a2228d4)

Reviewer 3: I see that this was annotated with a hash but not actually resolved.   Is there more todo?
